### PR TITLE
docs: add table with basic capabilities of different storage backends

### PR DESCRIPTION
**Pull request by davidbp**
This PR adds a table with storage backends and basic functionalities.

### CODE REVIEW

File: docs/advanced/document-store/index.md [hunk 1]
```diff
@@ -66,6 +66,29 @@ unexpected behaviors that can yield to, for example, inaccessible elements by po
...
+| [`Qdrant`](./qdrant/#qdrant)                                   | `DocumentArray(storage='qdrant')`        | ✅             | ✅                      | ❌             |
+| [`Annlite`](./annlite/#annlite)                                | `DocumentArray(storage='anlite')`        | ✅             | ✅                      | ✅             |
+| [`ElasticSearch`](./elasticsearch/#elasticsearch)              | `DocumentArray(storage='elasticsearch')` | ✅             | ✅                      | ✅             |
+
+
+Here we understand by 
+
+- **vector search**: perform approximate nearest neighbour search (or exact full scan search). The input is a vector. The result is a DocumentArray with the closest vectors to the query vector.
+
+- **vector search + filter**: perform approximate nearest neighbour search (or exact full scan search). The input is a vector and a filter. The result is a DocumentArray with the closest vectors to the query vector that verify the filter.
```
Comment (JoanFM) [inline docs/advanced/document-store/index.md:88]: the input is a `DocumentArray` with `embeddings not a vector

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docs/advanced/document-store/index.md (hunk 1 above)
Comment (davidbp) [reply docs/advanced/document-store/index.md]: makes sense, updated

File: docs/advanced/document-store/index.md [hunk 2]
```diff
@@ -133,6 +134,203 @@ da = DocumentArray(
 
 Using dataclass gives you better type-checking in IDE but requires an extra import; using dict is more flexible but can be error-prone. You can choose the style that fits best to your context.
 
+
+## Search summary
+
+DocArray supports multiple storage backends with different search features. The following table showcases relevant functionalities that are supported (✅) or not supported (❌) in DocArray depending on the backend:
+
+
+| Name                                                           | Construction                             | vector search | vector search + filter | filter|
```
Comment (JohannesMessner) [inline docs/advanced/document-store/index.md:143]:
```suggestion
| Name                                                           | Construction                             | fast vector search | vector search + filter | filter|
```
I think that gives more clarity without having to read the explanation

File: docs/advanced/document-store/index.md [hunk 3]
```diff
@@ -133,6 +134,203 @@ da = DocumentArray(
...
+| [`Sqlite`](./sqlite/#sqlite)                                   | `DocumentArray(storage='sqlite')`        | ❌             | ❌                      | ✅             | 
+| [`Weaviate`](./weaviate/#weaviate)                             | `DocumentArray(storage='weaviate')`      | ✅             | ✅                      | ✅             |
+| [`Qdrant`](./qdrant/#qdrant)                                   | `DocumentArray(storage='qdrant')`        | ✅             | ✅                      | ❌             |
+| [`Annlite`](./annlite/#annlite)                                | `DocumentArray(storage='anlite')`        | ✅             | ✅                      | ✅             |
+| [`ElasticSearch`](./elasticsearch/#elasticsearch)              | `DocumentArray(storage='elasticsearch')` | ✅             | ✅                      | ✅             |
+
+
+Here we understand by 
+
+- **vector search**: perform approximate nearest neighbour search (or exact full scan search). The input of the search function is a numpy array or a DocumentArray containing an embedding.
```
Comment (JohannesMessner) [inline docs/advanced/document-store/index.md:155]:
```suggestion
- **fast vector search**: perform approximate nearest neighbour search (or exact full scan search). The input of the search function is a numpy array or a DocumentArray containing an embedding. 
```

File: docs/advanced/document-store/index.md [hunk 4]
```diff
@@ -133,6 +134,203 @@ da = DocumentArray(
...
+| [`Qdrant`](./qdrant/#qdrant)                                   | `DocumentArray(storage='qdrant')`        | ✅             | ✅                      | ❌             |
+| [`Annlite`](./annlite/#annlite)                                | `DocumentArray(storage='anlite')`        | ✅             | ✅                      | ✅             |
+| [`ElasticSearch`](./elasticsearch/#elasticsearch)              | `DocumentArray(storage='elasticsearch')` | ✅             | ✅                      | ✅             |
+
+
+Here we understand by 
+
+- **vector search**: perform approximate nearest neighbour search (or exact full scan search). The input of the search function is a numpy array or a DocumentArray containing an embedding. 
+
+- **vector search + filter**: perform approximate nearest neighbour search (or exact full scan search). The input of the search function is a numpy array or a DocumentArray containing an embedding and a filter.
```
Comment (JohannesMessner) [inline docs/advanced/document-store/index.md:157]: It is still not 100% clear to me when this filtering happens

**Review by JohannesMessner — COMMENTED** [review]
I think this adds a lot of clarity.
It would be awesome (though probably a separate PR) if the same information could be found in the docstring of each storage backend docarray.

File: docs/advanced/document-store/index.md (hunk 3 above)
Comment (davidbp) [reply docs/advanced/document-store/index.md:155]: won't add 'fast' because it depends on the DocumentArray that you use!

**Review by JohannesMessner — APPROVED** [review]

File: docs/advanced/document-store/index.md [hunk 5]
```diff
@@ -133,6 +134,203 @@ da = DocumentArray(
...
+
+
+| Name                                                           | Construction                             | fast vector search | vector search + filter | filter|
+|----------------------------------------------------------------|------------------------------------------|---------------|------------------------|---------------|
+| [`None`](./../../../fundamentals/documentarray/#documentarray) | `DocumentArray()`                        | ✅             | ✅                      | ✅             |
+| [`Sqlite`](./sqlite/#sqlite)                                   | `DocumentArray(storage='sqlite')`        | ❌             | ❌                      | ✅             | 
+| [`Weaviate`](./weaviate/#weaviate)                             | `DocumentArray(storage='weaviate')`      | ✅             | ✅                      | ✅             |
+| [`Qdrant`](./qdrant/#qdrant)                                   | `DocumentArray(storage='qdrant')`        | ✅             | ✅                      | ❌             |
+| [`Annlite`](./annlite/#annlite)                                | `DocumentArray(storage='anlite')`        | ✅             | ✅                      | ✅             |
```
Comment (JohannesMessner) [inline docs/advanced/document-store/index.md:149]:
```suggestion
| [`Annlite`](./annlite/#annlite)                                | `DocumentArray(storage='annlite')`        | ✅             | ✅                      | ✅             |
```

File: docs/advanced/document-store/index.md [hunk 6]
```diff
@@ -133,6 +134,203 @@ da = DocumentArray(
...
 
+
+## Search summary
+
+DocArray supports multiple storage backends with different search features. The following table showcases relevant functionalities that are supported (✅) or not supported (❌) in DocArray depending on the backend:
+
+
+| Name                                                           | Construction                             | fast vector search | vector search + filter | filter|
+|----------------------------------------------------------------|------------------------------------------|---------------|------------------------|---------------|
+| [`None`](./../../../fundamentals/documentarray/#documentarray) | `DocumentArray()`                        | ✅             | ✅                      | ✅             |
```
Comment (JohannesMessner) [inline docs/advanced/document-store/index.md:145]: On second thought about the word "fast", does in memory docarray use some ANN technique or is it exhaustive search?

**Review by JohannesMessner — CHANGES_REQUESTED** [review]

**Review by JohannesMessner — APPROVED** [review]

**Review by JoanFM — APPROVED** [review]

**Pull request merged**